exercises/all-your-base/example.jl

```diff
@@ -0,0 +1,39 @@
+# Implement number base conversion
+
+function all_your_base(digits::Vector, input_base::Integer, output_base::Integer)
+    # error checking behavior
+    if input_base < 2
+        throw(ArgumentError("input base must be >= 2"))

```

Comment:
Two reasons, one is semi-practical, the other has to do with how the problem is structured.

1. Base one is effectively counting zero's (as the digits in the base are < the base).  This is pretty useless in practice, and so I think that's why...

2. In the problem specification (https://github.com/exercism/problem-specifications/blob/master/exercises/all-your-base/canonical-data.json), one of the tests is: `input base is one`; which should error.

I can relax this requirement, if you wish, but I think the problem spec is generally right in how most people think about bases.  One could also argue why use integers, as irrational bases are a thing (as an example: https://en.wikipedia.org/wiki/Golden_ratio_base) but I think that's taking it too far.

If you want me to change this, let me know and I shall.

Reply:
That's a good enough reason :) Please leave it as is.

Irrational bases might be a fun bonus or separate exercise. We could have the student yield an infinite (lazy) array on `all_your_base([1], φ, 10)` if we also returned the exponent.

Reply:
One other thing I thought of.  In [Positional Notation](https://en.wikipedia.org/wiki/Positional_notation) you have numbers in base **b** that are a linear combination of powers of **b**.  Taking **b** to be 1, and the *only* digit we can have in the base is 0 as it must be < **b**.  Given a sequence of digits `0000` we have:

(0 * 1^3) + (0 * 1^2) + (0 * 1^1) + (0 * 1^0) == 0

We *could* change the definition for the edge-case of base 1, so we *only* allow the digit 1, but then we have the problem on not being able to represent 0, as there is no finite - or infinite - sum of 1's that will add to zero.

It took me a while to think of this response.  It's still early for me and the coffee isn't flowing as fast just yet.

Reply:
I think unary notation is normally expressed as a string of `1`s anyway. And zero is the empty string. But you're right, it would need to be special cased and it is also useless.